downstairs/src/region.rs

```diff
@@ -714,10 +741,22 @@ impl Region {
         &self,
         writes: &[crucible_protocol::Write],
     ) -> Result<(), CrucibleError> {
+        let mut batched_writes: HashMap<usize, Vec<&crucible_protocol::Write>> =
+            HashMap::new();
```

Comment:
We should put a comment here describing why we build this HM (from the PR summary)

Reply:
added two comments in 8bac91d